Replace bash download via runCmd by simple command. Fix a few typos.

domcharrier opened the pull request:
RET uses array references and thus requires bash 4.3+.

The problem with the original bash installation procedure was
that this procedure itself required bash 4.3+ because of the "runCmd" function.
I simply use wget here now.

Minors:

* Fixed typos: *_DIRECTRORY -> *_DIRECTORY
* Added kernel version entry (3.10) for CentOS 7.7 in requirements/req.yml

Commit:
Replace bash download via runCmd by simple command. Fix a few typos.

domcharrier committed:
Remove support of CentOS 7.7 from reqirements YAML file.

domcharrier committed:
Reimplemented downloadFile routine as it is used four times in the code. Use it to download bash 4.3+.

src/update_kernel:31:    downloadFile "$__kernel_headers_all" "$5" "ERR" "Download kernel shared headers"
src/update_kernel:36:    downloadFile "$__kernel_headers" "$5" "ERR" "Download kernel headers"
src/update_kernel:41:    downloadFile "$__kernel_image" "$5" "ERR" "Download kernel image"
src/update_kernel:46:    downloadFile "$__kernel_modules" "$5" "WARN" "Download kernel modules"

domcharrier committed:
Minor.

reger-men commented:
Please use this function instead:

```
# Download file from URL
# Input $1: URL
# Input $2: Output Path
# Input $3: Error type [ERR (stop the execution), WARN (only show the warning an keep running)]
# Input $4: Message text
function downloadFile {
    local __err_type_downloadFile=$3
    local __cmd_ret_downloadFile
    local __msg_txt_downloadFile=${4:-"Download file"}

    # Read terminal output and pass it to loadBar
    function updatePercentage {
        local __download_per
        local __LoadBar_Pos=0
        local __succ_updatePercentage=1
        local __previous_value_updatePercentage="${__download_per}"

        while read __download_per
        do
            if [ "${__download_per}" != "${__previous_value_updatePercentage}" ]; then
                ((__LoadBar_Pos++)); __LoadBar_Pos=$((__LoadBar_Pos % ${#LOADBAR}))
                startLoadBar "${__msg_txt_downloadFile}: ${__download_per}" ${__LoadBar_Pos}
                __previous_value_updatePercentage="${__download_per}"
                __succ_updatePercentage=0
            fi
        done
        [ $__succ_updatePercentage -eq 0 ] && stopLoadBar "Download Finished!" || stopLoadBar
        return $__succ_updatePercentage;
    }

    # Download file and get the current download state
    su -p $SUDO_USER -c "wget --progress=dot -P '$2' '$1' 2>&1 | grep --line-buffered '[[:space:]][[:digit:]][[:digit:]]*%' | sed -u -e 's|.* \([0-9]\+[0-9]*\%\).*|\1|g'" | updatePercentage

    # Check the return state
    __cmd_ret_downloadFile="$?"
    checkReturn "$__cmd_ret_downloadFile" "wget -P $2 $1" "${__err_type_downloadFile}" "There was a problem downloading this package!"
}```

Commit:
Use reger-men's download function

Bash installation succeeds on CentOS 7.7 but a
load bar process is killed.

Error message:

```
src/functions: line 868: 10479 Killed                  loadBar "$1" "SUBSTEP" "$2"
```

Commit:
src/functions/checkBashVersion: Remove extra loadbar wrapped around call of downloadFile.

Commit:
src/functions/downloadFile: Update docu.

Commit:
Unify spelling of Tensorflow / ResNet-50.
